app.py

```
@@ -1609,6 +1672,91 @@ def api_backup_drive_configure():
+@api_admin_required
+def save_ddns_config():
+    try:
+        data = request.json
+        if not data:
+             return jsonify({'success': False, 'message': 'Invalid payload'}), 400
+             
+        if 'duckdns' in data:
+            duckdns = data['duckdns']
+            config_manager.set_value('ddns', 'duckdns_enabled', str(duckdns.get('enabled', False)).lower())
+            config_manager.set_value('ddns', 'duckdns_domain', duckdns.get('domain', ''))
+            if 'token' in duckdns and duckdns['token']:
```

Ruff (0.15.9)

[warning] Unnecessary key check before dictionary access

Replace with `dict.get`

(RUF019)